src/Microsoft.DotNet.Interactive.HttpRequest.Tests/ParserTests.cs

```diff
@@ -0,0 +1,645 @@
+
+            result.SyntaxTree.RootNode.ChildNodes.Should().ContainSingle<HttpRequestNode>().Which
+                .MethodNode.Should().BeNull();
+        }
+    }
+
+    public class Headers
+    {
+        [Fact]
+        public void headers_are_parsed_correctly()
+        {
+            var result = HttpRequestParser.Parse(
+                """
+        GET https://example.com HTTP/1.1
+        Accept: */*
+        Accept-Encoding : gzip, deflate, br
+        Accept-Language : en-US,en;q=0.9
+        ContentLength:7060
+        Cookie: expor=;HSD=Ak_1ZasdqwASDASD;SSID=SASASSDFsdfsdf213123;APISID=WRQWRQWRQWRcc123123;
+        """);
+
+            using var _ = new AssertionScope();
+
+            var headersNode = result.SyntaxTree.RootNode
+                  .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which
+                  .ChildNodes.Should().ContainSingle<HttpHeadersNode>().Which;
+
+            var headerNodes = headersNode.HeaderNodes.ToArray();
+            headerNodes.Should().HaveCount(5);
+
+            headerNodes[0].NameNode.Text.Should().Be("Accept");
+            headerNodes[0].SeparatorNode.Text.Should().Be(":");
+            headerNodes[0].ValueNode.Text.Should().Be("*/*");
+
+            headerNodes[1].NameNode.Text.Should().Be("Accept-Encoding");
+            headerNodes[1].SeparatorNode.Text.Should().Be(":");
+            headerNodes[1].ValueNode.Text.Should().Be("gzip, deflate, br");
+
+            headerNodes[2].NameNode.Text.Should().Be("Accept-Language");
+            headerNodes[2].SeparatorNode.Text.Should().Be(":");
+            headerNodes[2].ValueNode.Text.Should().Be("en-US,en;q=0.9");
+
+            headerNodes[3].NameNode.Text.Should().Be("ContentLength");
+            headerNodes[3].SeparatorNode.Text.Should().Be(":");
+            headerNodes[3].ValueNode.Text.Should().Be("7060");
+
+            headerNodes[4].NameNode.Text.Should().Be("Cookie");
+            headerNodes[4].SeparatorNode.Text.Should().Be(":");
+            headerNodes[4].ValueNode.Text.Should().Be("expor=;HSD=Ak_1ZasdqwASDASD;SSID=SASASSDFsdfsdf213123;APISID=WRQWRQWRQWRcc123123;");
+        }
+    }
+
+    public class Body
+    {
+        [Fact]
+        public void header_with_body_is_parsed_correctly()
+        {
+            var result = HttpRequestParser.Parse(
+                """
+        POST https://example.com/comments HTTP/1.1
+        Content-Type: application/xml
+        Authorization: token xxx
+
+        <request>
+            <name>sample</name>
+            <time>Wed, 21 Oct 2015 18:27:50 GMT</time>
+        </request>
+        """);
+
+            using var _ = new AssertionScope();
+
+            var requestNode = result.SyntaxTree.RootNode
+                  .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which;
+
+            requestNode.HeadersNode.HeaderNodes.Count.Should().Be(2);
+            requestNode.BodySeparatorNode.TextWithTrivia.Should().Be("\r\n");
```

Comment:
This assertion should be removed. It's overspecifying the test.

Reply:
I think it may be good to have a single test that validates the presence of body separator. 

Same also applies to the header separator - we currently assert the contents of the header separator (i.e., `:`) multiple times in the test that tests headers. It would be nice to have a single test that validates the presence of the header separator instead.